### DISCUSSION TITLE
Is there a function (or possible) to get the inverse of a JAX function?

### DISCUSSION THREAD

**Discussion by nasyxx** [jax-ml/jax#15446] [General]
We have the auto grad of a function; can we have the auto inverse of a function?

In this way, when we want to calculate the inverse function of `y = f(x) = sin (x ** 2 - 1 )`, we don't need to manually write `x = f^(-1)(x) = sqrt(arcsin(y) + 1)`, just an `inverse(f)(y)`.

**Comment by jakevdp** [comment]
JAX did have an experimental inverse transform at one point, but it was never fully implemented and has been removed.

If you want to see how it might look, the [custom jaxpr interpreter tutorial](https://jax.readthedocs.io/en/latest/notebooks/Writing_custom_interpreters_in_Jax.html#your-first-interpreter-invert) uses an `invert` transformation as its example application.

**Reply by mattjj** [reply]
I think there's an `oryx.core.inverse` function in [Oryx](https://github.com/jax-ml/oryx); see [Automatic function inversion in this colab](https://github.com/jax-ml/oryx/blob/main/oryx/examples/notebooks/a_tour_of_oryx.ipynb). (Developing that jaxpr interpreter tutorial is part of what led to Oryx!)

**Reply by nasyxx** [reply]
> If you want to see how it might look, the [custom jaxpr interpreter tutorial](https://jax.readthedocs.io/en/latest/notebooks/Writing_custom_interpreters_in_Jax.html#your-first-interpreter-invert) uses an `invert` transformation as its example application.

Thanks! Looks cool!

**Reply by nasyxx** [reply]
> I think there's an `oryx.core.inverse` function in [Oryx](https://github.com/jax-ml/oryx);

Nice!  I never expected such a function in such a probability library!

**Reply by krzysztofrusek** [reply]
Distrax also has an  inv transform https://github.com/deepmind/distrax/blob/b4d78b17ae4aaa3d25c8ca32430dfac8f826d299/distrax/_src/utils/transformations.py#L112
